dora-parser/src/parser.rs

```diff
@@ -1130,6 +1130,10 @@ impl<'a> Parser<'a> {
             self.expect_token(TokenKind::Eq)?;
             let expr = self.parse_expression()?;
 
+            Ok(Some(expr))
+        } else if self.token.is(TokenKind::LBrace) {
```

Comment:
Does this even need special handling? The block is already an expression and should be handles in the `self.parse_expression()` call above?

Reply:
Ugh ... yes, you are absolutely right. I'll remove everything but the new tests. Sorry!